pkg/controllermanager/controller/project/project_secret_activity_control.go

```diff
@@ -0,0 +1,127 @@
+func NewSecretActivityReconciler(gardenClient client.Client) reconcile.Reconciler {
+	return &projectSecretActivityReconciler{
+		gardenClient: gardenClient,
+	}
+}
+
+type projectSecretActivityReconciler struct {
+	gardenClient client.Client
+}
+
+func (r *projectSecretActivityReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+
+	secret := &corev1.Secret{}
+	if err := r.gardenClient.Get(ctx, client.ObjectKey{Namespace: request.Namespace, Name: request.Name}, secret); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Object is gone, stop reconciling")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
+	}
+
+	return r.reconcile(ctx, log, secret)
+}
+
+func (r *projectSecretActivityReconciler) reconcile(ctx context.Context, log logr.Logger, obj client.Object) (reconcile.Result, error) {
+	project, err := gutil.ProjectForNamespaceFromReader(ctx, r.gardenClient, obj.GetNamespace())
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// project is gone, nothing to do here
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("failed to get project for namespace %q: %w", obj.GetNamespace(), err)
+	}
+
+	if project.Status.LastActivityTimestamp != nil && obj.GetCreationTimestamp().UTC().Before(project.Status.LastActivityTimestamp.UTC()) {
+		// not the newest object in this project, nothing to do
+		return reconcile.Result{}, nil
+	}
+
+	timestamp := obj.GetCreationTimestamp()
+	if project.Status.LastActivityTimestamp == nil {
+		secrets := &corev1.SecretList{}
+		if err := r.gardenClient.List(
+			ctx,
+			secrets,
+			client.InNamespace(*project.Spec.Namespace),
+			client.MatchingLabels{v1beta1constants.LabelSecretBindingReference: "true"},
+		); err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to list secrets in project namespace %q: %w", *project.Spec.Namespace, err)
+		}
+		for _, secret := range secrets.Items {
+			if secret.GetCreationTimestamp().UTC().After(timestamp.UTC()) {
+				timestamp = secret.GetCreationTimestamp()
+			}
+		}
+	}
+
+	log.Info("Updating Project's lastActivityTimestamp", "lastActivityTimestamp", timestamp)
+	if err := updateStatus(ctx, r.gardenClient, project, func() {
+		project.Status.LastActivityTimestamp = &timestamp
+	}); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed updating Project's lastActivityTimestamp: %w", err)
+	}
+	return reconcile.Result{}, nil
+}
+
+func (c *Controller) updateSecretActivity(oldObj, newObj interface{}) {
+	newSecret, ok := newObj.(*corev1.Secret)
+	if !ok {
+		return
+	}
+	oldSecret, ok := oldObj.(*corev1.Secret)
+	if !ok {
+		return
+	}
+
+	// skip queueing if oldSecret has the label, which implies it already caused a LastActivityTimestamp update
```

Comment:
This assumption is problematic because controllers can miss events for arbitrary reasons and then this `Secret` will not be enqueued ever again.

Reply:
Then when we delete the last secretbinding for a secret, and the label is removed from the secret, it will cause an update of the lastActivityTimestamp.

Reply:
Also, updates are caused by finalizer calls, other label calls too..

Reply:
@rfranzke

Reply:
I don't see this code in the current version anymore, is there something left open to discuss?

Reply:
Ok, sorry. It was the condition,
```
if !metav1.HasLabel(newSecret.ObjectMeta, v1beta1constants.LabelSecretBindingReference) || 
	metav1.HasLabel(oldSecret.ObjectMeta,v1beta1constants.LabelSecretBindingReference )
```

Reply:
I want to change it back. Because I saw false timestamp updates because of this.